### ISSUE TITLE
Image upload (UploadMediaAsync) fails with ReasonPhrase "Forbidden"

### ISSUE THREAD

**Issue by jmnicolas90** [JoeMayo/LinqToTwitter#273]
Hello, 

I'm trying to create a simple bot that will post Wordpress posts to my Twitter account.
The tweets are composed of a post title, a link to the post and an image.
I manage to tweet text only but when I try to upload an image an exception is thrown with an empty message and ReasonPhrase	"Forbidden".

Here is the simplest code :

`var auth = new SingleUserAuthorizer
{
	CredentialStore = new SingleUserInMemoryCredentialStore
	{
		ConsumerKey = API_KEY,
		ConsumerSecret = API_SECRET,
		AccessToken = ACCESS_TOKEN,
		AccessTokenSecret = ACCESS_TOKEN_SECRET,
	}
};

await auth.AuthorizeAsync(); // not necessary?

var twitterCtx = new TwitterContext(auth);

var tweet = await twitterCtx.TweetAsync("test"); // OK, tweet is posted

var media = await twitterCtx.UploadMediaAsync(imgBlob, "image/jpg", "tweet_image"); // throws exception with reason "Forbidden"

// imgBlob is a alid byte[], if I write it to disk I get my original image back`

Is it a permission problem? I gave my app read,write and direct message permissions. 
I just subscribed to a dev account today, I have the "Essential Access" maybe I should apply for Elevated?

Any help would be appreciated, thank you.

**Comment by JeepNL** [comment]
You can't add an image to a Tweet after you posted it. 

Here's some sample code which should help you further:

https://github.com/JoeMayo/LinqToTwitter/blob/dfe9bf47de8b119bcb29260f38984e73f381d157/Samples/LinqToTwitter5/net48/CSharp/AspNetSamples/CoreDemo/Controllers/StatusDemosController.cs#L85

**Comment by jmnicolas90** [comment]
> You can't add an image to a Tweet after you posted it.

Yes, the code I posted is just an example, I wanted to show that I can post a text tweet.
If you delete the line where I post the the tweet, the image upload still throws an exception, so I won't be able to attach it to a latter tweet.

**Comment by JeepNL** [comment]
Look at this line in the example: `status tweet = await twitterCtx.TweetAsync(status, mediaIds);` This is how you should post a tweet with one or more images attached to it.

**Comment by JeepNL** [comment]
The previous link I posted was for LinqtoTwitter v5, below is for v6

https://github.com/JoeMayo/LinqToTwitter/blob/dfe9bf47de8b119bcb29260f38984e73f381d157/Samples/LinqToTwitter6/ASP.NET/LinqToTwitter.MVC.CSharp/LinqToTwitter.MVC.CSharp/Controllers/TweetDemosController.cs#L102

**Comment by jmnicolas90** [comment]
Yes I know, but to post a tweet with an image, I must first upload the image to get its media id.
It's the image upload that fails, before tweeting.

As per your example you first use ctx.UploadMediaAsync(...) before ctx.TweetMediaAsync(status, mediaIds);
I get my exception at ctx.UploadMediaAsync(...)

**Comment by JeepNL** [comment]
Ah, okay. Maybe this link helps > https://github.com/JoeMayo/LinqToTwitter/wiki/Uploading-Images-and-Video

**Comment by JeepNL** [comment]
Did you post [this question](https://stackoverflow.com/questions/71029763/twitter-api-error-request-failed-with-code-403-image-upload) at SO? 'Essential Access' let you only access V2 API end points, and in V2 image upload is not supported yet.
